parcels/field.py

```diff
@@ -887,59 +897,68 @@ def spatial_c_grid_interpolation2D(self, ti, z, y, x, time, particle=None, apply
         c2 = i_u._geodetic_distance(py[1], py[2], px[1], px[2], grid.mesh, np.dot(i_u.phi2D_lin(eta, 1.0), py))
         c3 = i_u._geodetic_distance(py[2], py[3], px[2], px[3], grid.mesh, np.dot(i_u.phi2D_lin(1.0, xsi), py))
         c4 = i_u._geodetic_distance(py[3], py[0], px[3], px[0], grid.mesh, np.dot(i_u.phi2D_lin(eta, 0.0), py))
-        if grid.zdim == 1:
-            if self.gridindexingtype == "nemo":
-                U0 = self.U.data[ti, yi + 1, xi] * c4
-                U1 = self.U.data[ti, yi + 1, xi + 1] * c2
-                V0 = self.V.data[ti, yi, xi + 1] * c1
-                V1 = self.V.data[ti, yi + 1, xi + 1] * c3
-            elif self.gridindexingtype in ["mitgcm", "croco"]:
-                U0 = self.U.data[ti, yi, xi] * c4
-                U1 = self.U.data[ti, yi, xi + 1] * c2
-                V0 = self.V.data[ti, yi, xi] * c1
-                V1 = self.V.data[ti, yi + 1, xi] * c3
-        else:
-            if self.gridindexingtype == "nemo":
-                U0 = self.U.data[ti, zi, yi + 1, xi] * c4
-                U1 = self.U.data[ti, zi, yi + 1, xi + 1] * c2
-                V0 = self.V.data[ti, zi, yi, xi + 1] * c1
-                V1 = self.V.data[ti, zi, yi + 1, xi + 1] * c3
-            elif self.gridindexingtype in ["mitgcm", "croco"]:
-                U0 = self.U.data[ti, zi, yi, xi] * c4
-                U1 = self.U.data[ti, zi, yi, xi + 1] * c2
-                V0 = self.V.data[ti, zi, yi, xi] * c1
-                V1 = self.V.data[ti, zi, yi + 1, xi] * c3
-        U = (1 - xsi) * U0 + xsi * U1
-        V = (1 - eta) * V0 + eta * V1
-        rad = np.pi / 180.0
-        deg2m = 1852 * 60.0
-        if applyConversion:
-            meshJac = (deg2m * deg2m * math.cos(rad * y)) if grid.mesh == "spherical" else 1
-        else:
-            meshJac = deg2m if grid.mesh == "spherical" else 1
-
-        jac = i_u._compute_jacobian_determinant(py, px, eta, xsi) * meshJac
-
-        u = (
-            (-(1 - eta) * U - (1 - xsi) * V) * px[0]
-            + ((1 - eta) * U - xsi * V) * px[1]
-            + (eta * U + xsi * V) * px[2]
-            + (-eta * U + (1 - xsi) * V) * px[3]
-        ) / jac
-        v = (
-            (-(1 - eta) * U - (1 - xsi) * V) * py[0]
-            + ((1 - eta) * U - xsi * V) * py[1]
-            + (eta * U + xsi * V) * py[2]
-            + (-eta * U + (1 - xsi) * V) * py[3]
-        ) / jac
-        if isinstance(u, da.core.Array):
-            u = u.compute()
-            v = v.compute()
+
+        def _calc_UV(ti, yi, xi):
```

Comment:
Should we move this function definition out of `spatial_c_grid_interpolation2D`?

Reply:
It's really only need for `spatial_c_grid_interpolation2D`, so I'd prefer to keep it here (for now)